flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/threadinfo/JobVertexThreadInfoTracker.java

```diff
@@ -251,7 +251,7 @@ private void triggerThreadInfoSampleInternal(
         return executionsWithGateways;
     }
 
-    private Map<TaskManagerLocation, ImmutableSet<ExecutionAttemptID>> groupExecutionsByLocation(
+    Map<TaskManagerLocation, ImmutableSet<ExecutionAttemptID>> groupExecutionsByLocation(
```

Comment:
Having to change the method visibility for testing is usually an indicator of testing against internal implementation. We should treat the testee as blackbox and test against its contract. In this case, the contract should be that `getVertexStats` should work properly when there's multiple executions for the same execution vertex.

Reply:
And in cases where there's no good way for testing without accessing internal methods, we should add an annotation `@VisibleForTesting`.